dask/dataframe/io/parquet/arrow.py

```diff
@@ -1238,110 +1077,65 @@ def _generate_dd_meta(cls, schema, index, categories, partition_info):
-            schema,
-            split_row_groups,
-            gather_statistics,
-            stat_col_indices,
-            filters,
-            categories,
-            partition_info,
-            data_path,
-            fs,
-            read_from_paths,
-            chunksize,
-            aggregation_depth,
-        )
+        return meta
 
     @classmethod
-    def _update_metadata_options(
-        cls,
-        gather_statistics,
-        split_row_groups,
-        metadata,
-        schema,
-        index_cols,
-        filters,
-        partition_info,
-        chunksize,
-        aggregation_depth,
-    ):
-        """Update read_parquet options given up-to-data metadata.
-
-        The primary focus here is `gather_statistics`. We want to
-        avoid setting this option to `True` if it is unnecessary.
-
-        This method is used by both `ArrowDatasetEngine`
-        and `ArrowLegacyEngine`.
+    def _construct_collection_plan(cls, dataset_info):
+        """pyarrow.dataset version of _construct_collection_plan
+        Use dataset_info to construct the general plan for
+        generating the output DataFrame collection.
+
+        The "plan" is essentially a list (called `parts`) of
+        information that is needed to produce each output partition.
+        After this function is returned, the information in each
+        element of `parts` will be used to produce a single Dask-
+        DataFrame partition (unless some elements of `parts`
+        are aggregated together in a follow-up step).
+
+        This method also returns ``stats`` (which is a list of
+        parquet-metadata statistics for each element of parts),
+        and ``common_metadata`` (which is a dictionary of kwargs
+        that should be passed to the ``read_partition`` call for
+        every output partition).
+
+        This method is overridden in `ArrowDatasetEngine`.
         """
 
+        # Collect necessary dataset information from dataset_info
+        ds = dataset_info["ds"]
+        fs = dataset_info["fs"]
+        filters = dataset_info["filters"]
+        split_row_groups = dataset_info["split_row_groups"]
+        gather_statistics = dataset_info["gather_statistics"]
+        chunksize = dataset_info["chunksize"]
+        aggregation_depth = dataset_info["aggregation_depth"]
+        index_cols = dataset_info["index_cols"]
+        schema = dataset_info["schema"]
+        partition_names = dataset_info["partition_names"]
+        partitioning_method = dataset_info["partitioning_method"]
+        partitions = dataset_info["partitions"]
+        categories = dataset_info["categories"]
+        has_metadata_file = dataset_info["has_metadata_file"]
+        valid_paths = dataset_info["valid_paths"]
+        metadata_task_size = dataset_info["metadata_task_size"]
+
+        # Check metadata_task_size setting
+        if metadata_task_size is None:
+            # Use 32 files per task by deault
+            metadata_task_size = 128
```

Comment:
Quick question: Is there a reason we need to assign `metadata_task_size = 128` like this, or could we just set that as the default in `read_parquet`?

Reply:
No - I'd probably prefer the default to be in read_parquet, but I wasn't sure if an engine-specific default might make sense.